### PR TITLE
webui: Use client side decorations for browser

### PR DESCRIPTION
**Pull request by halfline**
At the moment we show a titlebar with "Mozilla Firefox" in the title.

It looks much better to use client side decorations where the titlebar is rendered different and without Firefox branding.

This commit achieves that by:

1. Undoing the config override that forces the server side titlebar to be on
2. Putting MOZ_GTK_TITLEBAR_DECORATION=client in the environment so firefox (which is currently running as root segregated from the session) knows its okay to use client side decorations.

### CODE REVIEW

**Review by KKoukiou — APPROVED** [review]
Can you please add a screenshot before and after in the PR description?

**Comment by halfline** [comment]
Screenshot from 2023-08-21 09-19-53
Screenshot from 2023-08-21 09-16-07

**Review by vojtechtrefny — APPROVED** [review]

**Review by VladimirSlavik — APPROVED** [review]
Thank you!

**Comment by AdamWill** [comment]
Does this fix or improve the title of the window in the overview, if you open it? How about the icon?

**Comment by stransky** [comment]
[halfline](https://github.com/halfline) I wonder why do you need to set MOZ_GTK_TITLEBAR_DECORATION=client as it should be the default?

If Firefox is running with system titlebar enabled it means system environment auto-detection is broken or non-existent. Is that expected?

I mean it's okay to set MOZ_GTK_TITLEBAR_DECORATION=client but Firefox usually gets all config from system so if there's anything special you want to achieve (enabled system titlebar, different title etc.) there may be easier methods available than modify whole system setup for it.

**Comment by halfline** [comment]
> [halfline](https://github.com/halfline) I wonder why do you need to set MOZ_GTK_TITLEBAR_DECORATION=client as it should be the default?
> 
> If Firefox is running with system titlebar enabled it means system environment auto-detection is broken or non-existent. > Is that expected?

I think it's because it's running as root.  It may not be needed after https://github.com/rhinstaller/anaconda/pull/5058 but I haven't test that.

> there's anything special you want to achieve (enabled system titlebar, different title etc.) there may be easier methods available than modify whole system setup for it.

Actually if you have insight into less fragile ways to allow list external url handlers than editing the sqlite database (see https://github.com/rhinstaller/anaconda/pull/5059#issuecomment-1688921836 ), I'd be grateful.

**Comment by stransky** [comment]
> I think it's because it's running as root segregated from the rest of the session. It may not be needed after #5058 but I haven't test that.

I don't think root is related here. Firefox uses XDG_* env variables to get system setup. I think if you set 'XDG_CURRENT_DESKTOP=GNOME' you'll get the same Firefox look as in regular gnome session (CSD enabled, titlebar off).

**Comment by stransky** [comment]
> Actually if you have insight into less fragile ways to allow list external url handlers than editing the sqlite database (see [#5059 (comment)](https://github.com/rhinstaller/anaconda/pull/5059#issuecomment-1688921836) ), I'd be grateful.

I'm sure this can be handled somehow. Will look at it.

**Comment by halfline** [comment]
> I don't think root is related here. Firefox uses XDG_* env variables to get system setup. I think if you set 'XDG_CURRENT_DESKTOP=GNOME' you'll get the same Firefox look as in regular gnome session (CSD enabled, titlebar off).

Well it's not related to the uid of the running process directly, but escalating to root cleaves the app from aspects of the session and the environment (including `XDG_CURRENT_DESKTOP`), so that does explain it. We could make sure XDG_CURRENT_DESKTOP=GNOME bubbles through.  Does CSD get used on KDE as well? (just thinking about making sure things are futureproofed for the kde live images later on)

**Comment by stransky** [comment]
> Well it's not related to the uid of the running process directly, but escalating to root cleaves the app from aspects of the session and the environment (including `XDG_CURRENT_DESKTOP`), so that does explain it. We could make sure XDG_CURRENT_DESKTOP=GNOME bubbles through. Does CSD get used on KDE as well? (just thinking about making sure things are futureproofed for the kde live images later on)

Yes, Firefox uses CSD on all systems where XDG_CURRENT_DESKTOP is set, which includes KDE,Gnome,Mate,XFCE,Unity and so on. Non-decorated windows are used on twm and similar ones where XDG_CURRENT_DESKTOP is missing.

If XDG_CURRENT_DESKTOP is set it also removes system titlebar - not sure if you want it.
And if system titlebar is removed you have full control which titlebar buttons are shown in Firefox (I think there was an issue that Firefox shows a titlebar with close button).

**Comment by halfline** [comment]
alright, let's just let XDG_CURRENT_DESKTOP bubble through then.

**Comment by stransky** [comment]
> alright, let's just let XDG_CURRENT_DESKTOP bubble through then.

Okay. Feel free to ping me if you have any Firefox related issues.

**Comment by halfline** [comment]
okay i added an additional commit to hide the close button.

**Comment by halfline** [comment]
@VladimirSlavik time to land this?

**Comment by M4rtinK** [comment]
I have a solution with similar end result almost ready as part of #5112 which also handles how the application looks like in the Gnome Shell overview mode. Please don't merge this until after #5112 is merged, if still applicable, to avoid potential breakage. Thanks!

**Comment by halfline** [comment]
let's close this, now that https://github.com/rhinstaller/anaconda/pull/5112 has landed this isn't offering much I guess

**Pull request closed without merging**
